src/ztex_bcrypt.c

```diff
@@ -205,12 +206,24 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 	static unsigned char salt_buf[17]; // salt to send to device
 	BF_salt *BF_salt = salt->salt;
 
+	static int cmp_entries_excess = 0;
+	if (salt->count > bitstream.cmp_entries_max && !cmp_entries_excess) {
+		fprintf(stderr, "Warning: salt with %d hashes, onboard comparators "
+			"support up to %d hashes/salt, turned off\n",
+			salt->count, bitstream.cmp_entries_max);
+		jtr_device_list_set_app_mode(0x40);
```

Comment:
If `jtr_device_list_set_app_mode` isn't very expensive and supports being re-done multiple times, we could possibly do it per-salt whenever the new mode would be different from the previous one. This would allow to use on-board comparators for salts with up to 512 hashes even if some salt is above that threshold. It would also re-enable use of on-board comparators when the remaining hash count for the "problematic" salt(s) drops to 512 and below.

Reply:
Definitely a good idea, I'm including in TODO.
```jtr_device_list_set_app_mode``` would take extra USB transfer per fpga which sets mode of operation from Cypress controller using low-speed interface. It's possible to further optimize it, save USB transfers by introducing some ```SET_MODE``` packet which would be included in a stream of high-speed packet communication.